migrate.py

```diff
@@ -29,6 +30,53 @@
 PLUGIN_EXCEPTION_PATHS = {'modules': 'lib/ansible/modules', 'module_utils': 'lib/ansible/module_utils'}
 
 
+def _run_command(cmd=None, check_rc=True):
+    logger.debug(cmd)
+    if not isinstance(cmd, bytes):
+        cmd = cmd.encode('utf-8')
+    p = subprocess.Popen(
+        cmd,
+        shell=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE
+    )
+    (so, se) = p.communicate()
+
+    if check_rc and p.returncode != 0:
+        raise RuntimeError(se)
+
+    so = so.decode('utf-8')
+    se = se.decode('utf-8')
+
+    return (p.returncode, so, se)
+
+
+def run_command(cmd=None, check_rc=True):
+    (rc, so, se) = _run_command(cmd, check_rc)
+    return {
+        'rc': rc,
+        'so': so,
+        'se': se
+    }
+
+
+def checkout_repo(refresh=False):
+    releases_dir = os.path.join(VARDIR, 'releases')
+    devel_path = os.path.join(releases_dir, 'devel.git')
+
+    if refresh and os.path.exists(devel_path):
```

Comment:
If we aren't limited by outdates Pythons I'd prefer pathlib to be used instead.

Reply:
not what we should be focusing on, if the current functions can work with minor fix, i prefer to do that, we can change it later on once all the actual features are working